.github/workflows/build_main.yml

```diff
@@ -0,0 +1,26 @@
+name: build main
+on:
+  push:
+    branches-ignore:
+      - "python/*"
+      - "scala/*"
```

Comment:
do we need build_main and build_python differentiation given we only have a python build here?

Reply:
we may add other langs in the future, so while it adds some complexity, it isnt massive overhead, so I'd keep it this way